src/discord-cluster-manager/modal_runner_archs.py

```diff
@@ -0,0 +1,199 @@
+# This file contains wrapper functions for running
+# Modal apps on specific devices. We will fix this later.
+
+from modal_runner import app, cuda_image, run_cuda_script, run_pytorch_script
+
+
+# T4: sm_70 (CUDA 7.x, Maxwell Architecture)
+@app.function(
+    gpu="T4",
+    image=cuda_image,
+)
+def run_cuda_script_t4(
+    script_content: str,
+    reference_content: str = None,
+    submission_content: str = None,
+    timeout_seconds: int = 600,
+) -> tuple[str, float]:
+    return run_cuda_script(
+        script_content,
+        reference_content,
+        submission_content,
+        timeout_seconds,
+        arch=75,
+    )
+
+
+@app.function(
+    gpu="T4",
+    image=cuda_image,
+)
+def run_pytorch_script_t4(
+    script_content: str,
+    reference_content: str = None,
+    submission_content: str = None,
+    timeout_seconds: int = 600,
+) -> tuple[str, float]:
+    return run_pytorch_script(
+        script_content,
+        reference_content,
+        submission_content,
+        timeout_seconds,
+        arch=75,  # Targeting T4 (sm_75)
+    )
+
+
+# L4: sm_80 (L4 Tensor Core architecture)
+@app.function(
+    gpu="L4",
+    image=cuda_image,
+)
+def run_cuda_script_l4(
+    script_content: str,
+    reference_content: str = None,
+    submission_content: str = None,
+    timeout_seconds: int = 600,
+) -> tuple[str, float]:
+    return run_cuda_script(
+        script_content,
+        reference_content,
+        submission_content,
+        timeout_seconds,
+        arch=80,
+    )
+
+
+@app.function(
+    gpu="L4",
+    image=cuda_image,
+)
+def run_pytorch_script_l4(
+    script_content: str,
+    reference_content: str = None,
+    submission_content: str = None,
+    timeout_seconds: int = 600,
+) -> tuple[str, float]:
+    return run_pytorch_script(
+        script_content,
+        reference_content,
+        submission_content,
+        timeout_seconds,
+        arch=80,  # Targeting L4 (sm_80)
+    )
+
+
+# A100: sm_80 (Ampere architecture)
```

Comment:
make a dict for enum for SM to GPU mapping otherwise people will need to do a google search first